src/lib.rs

```diff
@@ -21,32 +21,40 @@ assert_eq!(leftpad(s, 7), "  blubb");
 use std::borrow::Cow;
 use std::borrow::Borrow;
 
-/// Pads a string to the given length `len` by inserting the character `pad_char` from the left.
+/// Pads a string to the given number of chars by inserting the character `pad_char` from the left.
 ///
-/// If the given string has a length longer or equal to the desired length, it will be
+/// If the given string has more than or exactly the desired number of codepoints, it will be
 /// returned as-is.
 ///
+/// Codepoints are not graphemes, so the result might always not be what a human would expect.
+///
 /// # Examples
 ///
 /// ```
 /// use left_pad::leftpad_with;
 ///
-/// assert_eq!(leftpad_with("blubb", 7, ' '), "  blubb");
-/// assert_eq!(leftpad_with("blubb", 7, '.'), "..blubb");
+/// assert_eq!(leftpad_with("blübb", 7, ' '), "  blübb");
+/// assert_eq!(leftpad_with("blübb", 7, '→'), "→→blübb");
 ///
-/// assert_eq!(leftpad_with("blubb", 5, ' '), "blubb");
-/// assert_eq!(leftpad_with("blubb", 3, ' '), "blubb");
+/// assert_eq!(leftpad_with("blübb", 5, ' '), "blübb");
+/// assert_eq!(leftpad_with("blübb", 3, ' '), "blübb");
 /// ```
-pub fn leftpad_with<'a, S>(string: S, len: usize, pad_char: char) -> Cow<'a, str>
+pub fn leftpad_with<'a, S>(string: S, codepoints: usize, pad_char: char) -> Cow<'a, str>
     where S: Into<Cow<'a, str>>
 {
     let cow = string.into();
 
-    if !(len > cow.len()) {
+    let cow_codepoints = {
+        let s: &str = cow.borrow();
+        s.len() - s.bytes()
+                   .filter(|b| (b >> 6) == 0b10u8 )
+                   .count()
+    };
```

Comment:
Eh, I don't think it will be *much* slower, so I'd prefer the `.chars().count()` here for readability.